doc: tools: add tools section and pages

greg-fer opened the pull request:
Added tools section and pages in documentation.
This adds readmies for chip-cert and spake2p to the doc pool.

gautesl commented on docs/conf.py:
```diff
@@ -32,6 +32,8 @@
     "examples/thermostat/nxp/linux-se05x/README.md",
     "examples/common/m5stack-tft/repo",
     "docs/guides/README.md",
+    "tools/memory/memdf/README.md",
+    "tools/memory/platform/README.md",
```
Should also add to the `external_content_contents` list
```py
    (MATTER_BASE, "src/tools/**/*.md"),
    (MATTER_BASE, "scripts/tools/**/*.md"),
```

greg-fer commented:
@andrei-menzopol , I've moved the NXP guide you added from platforms/nxp/doc to docs/guides and added it to the doc pool at https://project-chip.github.io/connectedhomeip-doc/guides/index.html , so the page will be visible there once this is merged. Could you check everything is OK and sign off this change? Thanks!

andrei-menzopol commented:
@greg-fer, looks good. Thank you.

LuDuda reviewed (approved):
Thank you Greg!
